mlflow/langchain/utils.py

```diff
@@ -243,27 +243,31 @@ def _get_supported_llms():
     import langchain.chat_models
     import langchain.llms
 
-    llms = {langchain.llms.openai.OpenAI, langchain.llms.huggingface_hub.HuggingFaceHub}
+    supported_llms = set()
 
-    if hasattr(langchain.llms, "Databricks"):
-        llms.add(langchain.llms.Databricks)
+    def try_adding_llm(module, class_name):
+        if hasattr(module, class_name):
+            supported_llms.add(getattr(module, class_name))
 
-    if hasattr(langchain.llms, "Mlflow"):
-        llms.add(langchain.llms.Mlflow)
-
-    if hasattr(langchain.chat_models, "ChatDatabricks"):
-        llms.add(langchain.chat_models.ChatDatabricks)
+    def safe_import_and_add(module_name, class_name):
+        """Add conditional support for `partner` and `community` APIs in langchain"""
+        try:
+            module = __import__(module_name, fromlist=[class_name])
```

Comment:
Can we use `importlib.import_module` here?

Reply:
good call